src/range.rs

```diff
@@ -493,6 +493,11 @@ impl<V: Ord + Clone> Range<V> {
         }
         Self { segments }.check_invariants()
     }
+
+    /// Iterate over the parts of the range.
+    pub fn iter(&self) -> impl Iterator<Item = &(Bound<V>, Bound<V>)> {
```

Comment:
Can this be `Item = (&Bound<V>, &Bound<V>)`, so that our API does not require us to store this in a tuple?

Reply:
I've checked with uv, we can :+1: